### DISCUSSION TITLE
Welcome to palworld-save-pal Discussions!

### DISCUSSION THREAD

**Discussion by oMaN-Rod** [oMaN-Rod/palworld-save-pal#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rquaale** [comment]
So whenever i try to load in the .zip files it gives me the following error:
Error processing the zip file: Zip file does not contain 'Level.sav', available files: ['Players/', 'Players/00000000000000000000000000000001.sav', 'Level.sav']
Any idea why or how to fix?

**Reply by oMaN-Rod** [reply]
Check out the first video in the [Usage Guide](https://github.com/oMaN-Rod/palworld-save-pal/wiki/%F0%9F%8E%AE-Usage)

**Comment by jealousinterestingly** [comment]
A lot of AVs flag the zip as containing a trojan

**Reply by oMaN-Rod** [reply]
Compiled Python often triggers antivirus alerts because packaging tools bundle the Python interpreter with your code, which looks suspicious. Some malware uses Python, so antivirus software is extra cautious.

In this case it's a false positive, but if anyone is uncomfortable with it; compiling from source, using docker, or running the app directly with python are all options.

**Reply by jealousinterestingly** [reply]
Oh I see, didn't know that

**Comment by Enadddd** [comment]
Hi. I have followed all the steps but stuck at unpacking my zip files.  Here's the error message:

Traceback (most recent call last):
  File "palworld_save_pal\ws\manager.py", line 34, in process_message
  File "palworld_save_pal\ws\dispatcher.py", line 19, in dispatch
  File "palworld_save_pal\ws\handlers\save_file_handler.py", line 114, in load_zip_file_handler
  File "palworld_save_pal\state.py", line 25, in process_save_files
  File "palworld_save_pal\save_file\save_file.py", line 262, in load_sav_files
  File "palworld_save_tools\gvas.py", line 131, in read
  File "palworld_save_tools\archive.py", line 386, in properties_until_end
  File "palworld_save_tools\archive.py", line 399, in property
  File "palworld_save_tools\archive.py", line 534, in struct
  File "palworld_save_tools\archive.py", line 561, in struct_value
  File "palworld_save_tools\archive.py", line 386, in properties_until_end
  File "palworld_save_tools\archive.py", line 496, in property
  File "palworld_save_tools\archive.py", line 518, in prop_value
  File "palworld_save_tools\archive.py", line 561, in struct_value
  File "palworld_save_tools\archive.py", line 386, in properties_until_end
  File "palworld_save_tools\archive.py", line 475, in property
  File "palworld_save_tools\archive.py", line 575, in array_property
  File "palworld_save_tools\archive.py", line 561, in struct_value
  File "palworld_save_tools\archive.py", line 386, in properties_until_end
  File "palworld_save_tools\archive.py", line 396, in property
  File "palworld_save_pal\save_file\item_container_slot.py", line 17, in decode
  File "palworld_save_pal\save_file\item_container_slot.py", line 32, in decode_bytes
  File "palworld_save_tools\archive.py", line 333, in float
struct.error: unpack requires a buffer of 4 bytes

**Reply by oMaN-Rod** [reply]
What game version are you on?

**Reply by Enadddd** [reply]
currently playing on a cracked, on the bottom right it shows v0.3.6.57672

**Reply by oMaN-Rod** [reply]
[This](https://github.com/oMaN-Rod/palworld-save-pal/releases/tag/v0.4.2) version should work, v0.4.3+ are not backwards compatible with gave versions below 0.3.7.x.

**Reply by Enadddd** [reply]
Thank you! Much appreciated for the help.

**Comment by nalira90** [comment]
Amazing software, but I have one question which is; is there a possibility to change gui number of the players so I can keep my local co op player when I've transfered it over to a server? Much appreciated!

**Reply by oMaN-Rod** [reply]
My bad for missing this! Depending on how the update goes tomorrow this functionality will be available in the next 1 or 2 releases, I'm working on adding gamepass support, converting gamepass to steam and vice versa, converting from coop to dedicated and vice versa, transferring players and a few other features.

**Comment by Zappli45** [comment]
Currently not compatible with Feybreak Update.

Gotten "EOF not reached" errors after a new save point with the new update. Includes fresh worlds too.

**Comment by iSKL001** [comment]
Hi, I tried using the newest release and I have the latest 0.4.12 game version but it shows this message when opening PSP.exe, and when I look at background process I find it running but not working.
![image](https://github.com/user-attachments/assets/892143a4-c596-4a6c-b6f1-ff7323c3cace)

![image](https://github.com/user-attachments/assets/c7dcd729-0285-48b5-9e92-08f3ea1c03f0)

**Reply by oMaN-Rod** [reply]
Right click on the zip before extracting and select _Properties_, check _Unblock_ then click _Apply_. Extract as normal and start `PSP.exe`
 
![image](https://github.com/user-attachments/assets/e26ba5f8-ea7e-4924-aaba-bad3f9b64700)

**Reply by iSKL001** [reply]
works,thanks. any Idea on how to resize the ui to make it easier to choose the presets for passives/skills, on earlier versions I could do so by using ctrl + mouse wheel.

**Reply by iSKL001** [reply]
nevermind, using ctrl + [ +/- ] works

**Comment by Pandaboisenpai** [comment]
Hi i was just wondering how soon is gamepass saves gonna be just got my pc and have been playing on my xbox since launch thank you for everything you've done

**Reply by oMaN-Rod** [reply]
[This](https://github.com/oMaN-Rod/palworld-save-pal/tree/v0.8) branch implements gamepass support, you can build it from source using [this](https://github.com/oMaN-Rod/palworld-save-tools/tree/v0.4.11) fork of palworld-save-tools (upstream dependency that hasn't been updated for v0.4.11). If your looking for a compiled release you can join the discord and grab it from the #gamepass-beta channel, asking for help from the community to validate before officially releasing it.

**Comment by Tsuto76** [comment]
Hello, Thank you for releasing this great tool. However, I am currently experiencing an issue. When I edit my save data, it resets my friends' map unlocks and quest progress. My player level and companions are not reset. There is no issue on the host side. Is this because when I export my save data, it is saved as a separate save data? Please let me know if you have a solution.

**Reply by oMaN-Rod** [reply]
Thanks for reporting this, a few others have also shared this issue as well. Going to look into it, would also like to add a map unlock feature so this issue might be a blessing in disguise lol.

**Comment by am3692** [comment]
Hi, I am getting this error when trying to edit a base pal in the newest verison

`
2 validation errors for UpdateSaveFileMessage
data.modified_pals.4e7f52c3-4ed8-c85a-fbbc-6e87dd8078a0.rank
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='[object Object]1', input_type=str]
    For further information visit https://errors.pydantic.dev/2.10/v/int_parsing
data.modified_pals.4e7f52c3-4ed8-c85a-fbbc-6e87dd8078a0.max_hp
  Input should be a valid integer [type=int_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.10/v/int_type

Traceback (most recent call last):
  File "palworld_save_pal\ws\manager.py", line 34, in process_message
  File "palworld_save_pal\ws\dispatcher.py", line 18, in dispatch
  File "pydantic\main.py", line 214, in __init__
pydantic_core._pydantic_core.ValidationError: 2 validation errors for UpdateSaveFileMessage
data.modified_pals.4e7f52c3-4ed8-c85a-fbbc-6e87dd8078a0.rank
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='[object Object]1', input_type=str]
    For further information visit https://errors.pydantic.dev/2.10/v/int_parsing
data.modified_pals.4e7f52c3-4ed8-c85a-fbbc-6e87dd8078a0.max_hp
  Input should be a valid integer [type=int_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.10/v/int_type
`

**Reply by oMaN-Rod** [reply]
Do you remember which pal you were editing?

**Reply by am3692** [reply]
Lyleen

**Reply by oMaN-Rod** [reply]
Thanks for the info, found the issue, I will release a new version today.

**Comment by duugalas** [comment]
I cloned a xenolord a few times and gave each one different skills. Back in the game, none of the clones showed up in the palbox. Is there an issue with cloning xenolord, or an issue with the game not allowing that pal to have passive skills like: swift, runner, eternal engine, or infinite stamina? I used the PSP-windows-standalone-0.8.17, steam option, to edit  an xbox server level.sav file. Lots of other new and cloned pals worked just fine. Lots of other player items were added and worked just fine too. Only issue was with the Xenolord. But other edits that were done simultaneously with the xenolord clones did not show up. Almost as if the server rejected the level.sav file and reverted to the latest backup. Curious if anyone else has had issues with the xenolord pal. -Thanks
